packages/react-aria-components/src/RadioGroup.tsx

```diff
@@ -143,7 +143,11 @@ function RadioGroup(props: RadioGroupProps, ref: ForwardedRef<HTMLDivElement>) {
 function Radio(props: RadioProps, ref: ForwardedRef<HTMLInputElement>) {
   let state = React.useContext(InternalRadioContext)!;
   let domRef = useObjectRef(ref);
-  let {inputProps, isSelected, isDisabled, isPressed: isPressedKeyboard} = useRadio(props, state, domRef);
+  let {inputProps, isSelected, isDisabled, isPressed: isPressedKeyboard} = useRadio({
+    ...props,
+    // ReactNode type doesn't allow function children.
+    children: props.children != null
```

Comment:
This introduces a bug. If `props.children` is null or undefined, `useRadio()` gets `children` as `false`. When `useRadio()` computes `hasChildren` the [`false != null`](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/radio/src/useRadio.ts#L49) becomes true. 
Also done with Switch below.

Reply:
Great catch!